src/GafferArnoldUI/ArnoldLightVisualiser.cpp

```diff
@@ -0,0 +1,399 @@
+{
+	ShaderNetworkPtr oslNetwork = shaderNetwork->copy();
+
+	oslNetwork->setOutput( output );
+	ShaderNetworkAlgo::removeUnusedShaders( oslNetwork.get() );
+
+	bool hasUnconvertedArnoldShaders = false;
+
+	InternedString fallbackImageHandle;
+
+	for( const auto &s : oslNetwork->shaders() )
+	{
+		const Shader *shader = s.second.get();
+		if( shader->getType().substr( 0, 3 ) == "ai:" )
+		{
+			std::string outputName;
+			ShaderPtr standinShader = oslStandin( shader, outputName );
+			if( !standinShader )
+			{
+				hasUnconvertedArnoldShaders = true;
+				continue;
+			}
+
+			if( shader->getName() == "image" )
+			{
+				fallbackImageHandle = s.first;
+			}
+
+			oslNetwork->setShader( s.first, std::move( standinShader ) );
+			remapOutputConnections( s.first, "", s.first, outputName, oslNetwork.get() );
+		}
+	}
+
+	if( hasUnconvertedArnoldShaders && fallbackImageHandle != "" )
+	{
+		/// TODO: Tell the user which location this is
+		msg( Msg::Warning, "ArnoldLightVisualiser", "Unsupported shaders in network, falling back on " + fallbackImageHandle.string() );
+
+		ShaderNetworkPtr minimalNetwork = new ShaderNetwork();
+		minimalNetwork->addShader( "image", oslNetwork->getShader( fallbackImageHandle ) );
+		minimalNetwork->setOutput( ShaderNetwork::Parameter( "image", "out" ) );
+
+		oslNetwork = minimalNetwork;
+	}
+
+	return oslNetwork;
+}
+
+//////////////////////////////////////////////////////////////////////////
+// ArnoldLightVisualiser implementation
+//////////////////////////////////////////////////////////////////////////
+
+class GAFFERSCENEUI_API ArnoldLightVisualiser : public GafferSceneUI::StandardLightVisualiser
+{
+
+	public :
+
+		IE_CORE_DECLAREMEMBERPTR( ArnoldLightVisualiser )
+
+		ArnoldLightVisualiser();
+		~ArnoldLightVisualiser() override;
+
+	protected :
+
+		IECore::CompoundDataPtr surfaceTexture( const std::string &type, const IECore::InternedString &attributeName, const IECoreScene::ShaderNetwork *shaderNetwork, const IECore::CompoundObject *attributes ) const override;
+
+	private :
+
+		static LightVisualiser::LightVisualiserDescription<ArnoldLightVisualiser> g_description;
+
+};
+
+IE_CORE_DECLAREPTR( ArnoldLightVisualiser )
+
+
+IECoreGLPreview::LightVisualiser::LightVisualiserDescription<ArnoldLightVisualiser> ArnoldLightVisualiser::g_description( "ai:light", "*" );
+
+
+ArnoldLightVisualiser::ArnoldLightVisualiser()
+{
+}
+
+ArnoldLightVisualiser::~ArnoldLightVisualiser()
+{
+}
+
+IECore::CompoundDataPtr ArnoldLightVisualiser::surfaceTexture( const std::string &type, const IECore::InternedString &attributeName, const IECoreScene::ShaderNetwork *shaderNetwork, const IECore::CompoundObject *attributes ) const
+{
+	const ShaderNetwork::Parameter &output = shaderNetwork->getOutput();
+	if( !output )
+	{
+		return nullptr;
+	}
+
+	// Look for a 'color' input.
+	// TODO: Work out the right signature for surfaceTexture to let us look this up
```

Comment:
@johnhaddon It feels like if we get the signature of this right, this would be easier. Be good to chat when you have a sec.

Reply:
<note to self> pass `attirbutes`, `shaderNetwork` and `maxResolution`

Reply:
Fixed in https://github.com/GafferHQ/gaffer/commit/8916d9093f2206136577a8b2bceb1a961d0973c6